pkg/adapter/handler.go

```diff
@@ -40,4 +41,27 @@ type (
 		// report an error and stop serving runtime traffic to the particular Handler.
 		Build(config Config, env Env) (Handler, error)
 	}
+
+	// Builder2 represents a factory of handlers. Adapters register builders with Mixer
+	// in order to allow Mixer to instantiate handlers on demand.
+	//
+	// For a given builder, Mixer calls the various template-specific configuration methods
+	// and once done then Mixer calls the Validate followed by the Build method. The Build method
```

Comment:
Mixer calls the SetAdapterConfig followed by template-specific Set methods and once done then Mixer ..